core/src/main/res/values-night/color.xml

```diff
@@ -13,4 +13,8 @@
   <color name="color_on_primary">@color/white</color>
   <color name="color_on_surface">@color/white</color>
   <color name="color_on_error">@color/white</color>
+
+  <color name="text_primary">@color/white</color>
```

Comment:
text_primary is identical to `?colorOnSurface`, get rid of it and just use that. Infact you don't even need to specify it because that is the default already used for all textViews